lib/class-wp-block-patterns-registry.php

```diff
@@ -116,6 +116,45 @@ public static function get_instance() {
 	}
 }
 
+/**
+ * Registers a new pattern.
+ *
+ * @param string $pattern_name       Pattern name including namespace.
+ * @param array  $pattern_properties Array containing the properties of the pattern.
+ *
+ * @return boolean True if the pattern was registered with success and false otherwise.
+ */
+function register_block_pattern( $pattern_name, $pattern_properties ) {
+	return WP_Block_Patterns_Registry::get_instance()->register( $pattern_name, $pattern_properties );
+}
+
+/**
+ * Unregisters a pattern.
+ *
+ * @param string $pattern_name       Pattern name including namespace.
+ *
+ * @return boolean True if the pattern was unregistered with success and false otherwise.
+ */
+function unregister_block_pattern( $pattern_name ) {
+	return WP_Block_Patterns_Registry::get_instance()->unregister( $pattern_name );
+}
+
+/**
+ * Class used for interacting with patterns.
+ */
+class WP_Patterns_Registry {
+
+	/**
+	 * Utility method to retrieve the main instance of the class.
+	 *
+	 * @return WP_Block_Patterns_Registry The main instance.
+	 */
+	public static function get_instance() {
```

Comment:
_Technically_ one could have created their own `WP_Patterns_Registry` via `new WP_Patterns_Registry`. In fact, if we had any unit tests for this, I assume it would have been done this way. We don't currently deprecate any of the instance members, so custom constructed objects won't proxy through to the new class. I guess we could choose between: (a) Consider it unlikely and accept it, (b) add deprecations for all instance members.

Reply:
>(a) Consider it unlikely and accept it 

I'm leaning towards this because it's useless since it won't be even passed to the editor?